sqlglot/expressions.py

```diff
@@ -547,7 +555,7 @@ def sql(self, dialect: DialectType = None, **opts) -> str:
 
         return Dialect.get_or_raise(dialect).generate(self, **opts)
 
-    def transform(self, fun, *args, copy=True, **kwargs):
+    def transform(self, fun: t.Callable, *args: t.Any, copy: bool = True, **kwargs) -> Expression:
```

Comment:
ditto about more specific callable

Reply:
this can't be done, because it accepts kwargs